services/sysinfo/server/sysinfo.go

```diff
@@ -115,6 +113,21 @@ func (s *server) Dmesg(req *pb.DmesgRequest, stream pb.SysInfo_DmesgServer) erro
 	return nil
 }
 
+func normalizeTimeout(pbTimeout *durationpb.Duration) time.Duration {
+	if pbTimeout == nil {
+		return pb.MinDmesgTimeout
+	}
+
+	timeout := pbTimeout.AsDuration()
+	if timeout > pb.MaxDmesgTimeout {
```

Comment:
Shouldn't we throw an error if the requested timeout is out of range?

Reply:
Thanks, replaced normalization with error instead, it will still default to min value for nil param value (older client, newer server) but should throw an error on any value that is out of bounds.